src/__tests__/lib/utils/SynapseClient.test.ts

```diff
@@ -0,0 +1,230 @@
+import {
+  BackendDestinationEnum,
+  getEndpoint,
+} from '../../../lib/utils/functions/getEndpoint'
+import {
+  AsynchronousJobStatus,
+  PaginatedResults,
+} from '../../../lib/utils/synapseTypes'
+import { SynapseClient } from '../../../lib/utils'
+import { FunctionReturningPaginatedResults } from '../../../lib/utils/SynapseClient'
+import { rest, server } from '../../../mocks/msw/server'
+import { ASYNCHRONOUS_JOB_TOKEN } from '../../../lib/utils/APIConstants'
+
+describe('SynapseClient tests', () => {
+  describe('SynapseClient integration tests', () => {
```

Comment:
Use MSW to have integration tests without relying on the service to be up/configured in a certain way